codec-http2/src/test/java/io/netty/handler/codec/http2/Http2ConnectionHandlerTest.java

```diff
@@ -307,6 +308,46 @@ public void connectionErrorShouldStartShutdown() throws Exception {
     }
 
     @Test
+    public void serverShouldSend431WhenHeadersAreTooLarge() throws Exception {
+        handler = newHandler();
+        Http2Exception e = new Http2Exception.DecodeHeaderListSizeException(STREAM_ID, Http2Error.FRAME_SIZE_ERROR,
```

Comment:
Why `FRAME_SIZE_ERROR` here IIRC `PROTOCOL_ERROR` is more appropriate for this condition.